### DISCUSSION TITLE
Material TextTheme override

### DISCUSSION THREAD

**Discussion by itsezlife** [nank1ro/flutter-shadcn-ui#118] [Ideas]
Hello there! Really enjoy playing around with shadcn_ui package. 

I have one idea to consider. It'd be great to have an ability to override value to the Material text theme, not only for ShadTextTheme. I feel more comfortable using default TextTheme over ShadTextTheme. However, I don't know how to override default material TextTheme.

If there is an option, I'd be glad if you can point me out. If not, I'd glad if you can consider adding this feature.

Here is what can be done:
1. Add materialTextTheme optional field in `ShadBaseTheme`.
2. Add materialTextTheme optional field in `ShadThemeData`.
3. Inside of the `app.dart` file modify `materialTheme` function, line 638. Here you can check if there is an optional materialTextTheme from `theme` and if it is not null use it's value over `materialTheme` or you can merge the default one with provided(but I guess it's better not to merge).
4. And it's done, I guess. This way I can provide my own TextTheme and use Material TextTheme instead of ShadTextTheme.

**Comment by nank1ro** [comment]
Why didn't you use
```dart
materialThemeBuilder: (context, theme) {
        return theme.copyWith(
          textTheme: TextTheme(...),
        );
      },

```

**Reply by itsezlife** [reply]
Oh, didn't notice it. Thanks!
